[doc] add example of binding IP to ingress controller (#7758)
deckhouse/deckhouse: ee/modules/380-metallb/docs/EXAMPLES.md

title: "The metallb module: examples"
---

Metallb can be used in Static (Bare Metal) clusters when you can't order a load balancer from a cloud provider. Metallb can work in L2 or BGP modes. 

## Example of Metallb usage in L2 mode.

{% raw %}

> We will create an Ingress Controller with `LoadBalancer` inlet. And we will also expose standalone Nginx web server using a Service with the `LoadBalancer` type.

Reviewer: ```suggestion
The example shows the creation of an Ingress controller with an inlet `LoadBalancer`, as well as providing access to a separately running Nginx web server using a service with the type `LoadBalancer'.
```

> First, you have to decide, which NodeGroups will be used to deploy applications that have to be exposed by the LoadBalancer service.

Reviewer: ```suggestion
You have to decide, which NodeGroups will be used to deploy applications that have to be exposed by the `LoadBalancer` service.
```

title: "The metallb module: examples"
---

> Metallb can be used in Static (Bare Metal) clusters when you can't order a load balancer from a cloud provider. Metallb can work in L2 or BGP modes. 

Reviewer 2: ```suggestion
Metallb can be used in Static (Bare Metal) clusters when you can't order a load balancer from a cloud provider. Metallb can work in L2 or BGP modes.
```

   ```yaml
   apiVersion: deckhouse.io/v1
   kind: IngressNginxController
   metadata:
     name: main
   spec:
     ingressClass: nginx
     inlet: LoadBalancer
     nodeSelector:
       node-role.deckhouse.io/frontend: ""
     tolerations:
     - effect: NoExecute
       key: dedicated.deckhouse.io
       value: frontend
   ```

1. Проверьте, что сервис с типом `LoadBalancer` создан в _Namespace_ `d8-ingress-nginx`:

   ```shell
   kubectl -n d8-ingress-nginx get svc main-load-balancer
   ```

   Пример вывода:

   ```shell
   $ kubectl -n d8-ingress-nginx get svc main-load-balancer 
   NAME                 TYPE           CLUSTER-IP       EXTERNAL-IP       PORT(S)                      AGE
   main-load-balancer   LoadBalancer   10.222.255.194   192.168.199.100   80:30236/TCP,443:32292/TCP   30s
   ```

1. Проверьте, что Ingress-контроллер доступен по внешнему IP-адресу.

   Пример:

   ```console
   $ curl -s -o /dev/null -w "%{http_code}" 192.168.199.100
   404
   ```

1. Предоставьте доступ к веб-серверу Nginx на порту `8080`:

   ```shell
   kubectl create deploy nginx --image=nginx
   kubectl create svc loadbalancer nginx --tcp=8080:80
   ```

> 1. Проверьте, что сервис создан:

Reviewer 3: ```suggestion
1. Verify that the service has been created:
```

   ```shell
   kubectl get svc nginx
   ```

>    Пример вывода:

Reviewer 3: ```suggestion
   Your output should look something like this:
```

   ```shell
   $ kubectl get svc nginx
   NAME    TYPE           CLUSTER-IP     EXTERNAL-IP       PORT(S)          AGE
   nginx   LoadBalancer   10.222.9.190   192.168.199.101   8080:31689/TCP   3m11s
   ```

> 1. Проверьте доступ к приложению.
> 
>    Пример:

Reviewer 3: ```suggestion
1. Check if the application is accessible.

   Example:
```

```

{% endraw %}

> ## Example of binding a MetalLB IP address to an Ingress controller

Reviewer 2: It is an example for BGP mode, [we are working](https://github.com/deckhouse/deckhouse/pull/7752) on a new inlet, L2LoadBalancer. Not sure if we need IngressNginxController example here.
